source/Calamari/Integration/AppSettingsJson/AppSettingsJsonGenerator.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.IO;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Octostache;
+
+namespace Calamari.Integration.AppSettingsJson
+{
+    public class AppSettingsJsonGenerator : IAppSettingsJsonGenerator
+    {
+        const string KeyDelimiter = ":";
+        
+        public void Generate(string appSettingsFilePath, VariableDictionary variables)
+        {
+            var root = LoadJson(appSettingsFilePath);
+
+            var names = variables.GetNames();
+            names.Sort(StringComparer.OrdinalIgnoreCase);
+
+            foreach (var name in names)
+            {
+                if (name.StartsWith("Octopus", StringComparison.OrdinalIgnoreCase))
```

Comment:
I'm still not entirely convinced that we want to dump _all_ variables to json. but for want of a better process then this will do.